dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/ManagePermissionsPage.java

```diff
@@ -470,21 +472,33 @@ public void assignRole(ActionEvent evt) {
      * @param ra   The {@code RoleAssignee} to be notified.
      * @param type The type of notification.
      */
-    private void notifyRoleChange(RoleAssignee ra, UserNotification.Type type) {
+    private void notifyRoleChange(RoleAssignee ra, NotificationType type) {
         if (ra instanceof AuthenticatedUser) {
-            userNotificationService.sendNotification((AuthenticatedUser) ra, new Timestamp(new Date().getTime()), type, dvObject.getId());
+            userNotificationService.sendNotification((AuthenticatedUser) ra, new Timestamp(new Date().getTime()), type, dvObject.getId(), determinateObjectType(dvObject));
         } else if (ra instanceof ExplicitGroup) {
             ExplicitGroup eg = (ExplicitGroup) ra;
             Set<String> explicitGroupMembers = eg.getContainedRoleAssgineeIdentifiers();
             for (String id : explicitGroupMembers) {
                 RoleAssignee explicitGroupMember = roleAssigneeService.getRoleAssignee(id);
                 if (explicitGroupMember instanceof AuthenticatedUser) {
-                    userNotificationService.sendNotification((AuthenticatedUser) explicitGroupMember, new Timestamp(new Date().getTime()), type, dvObject.getId());
+                    userNotificationService.sendNotification((AuthenticatedUser) explicitGroupMember, new Timestamp(new Date().getTime()), type, dvObject.getId(), determinateObjectType(dvObject));
                 }
             }
         }
     }
 
+    private NotificationObjectType determinateObjectType(DvObject dvObject) {
```

Comment:
Although 'determinate' isn't strictly speaking incorrect I feel like 'determine' or simple 'getObjectType' would be more accurate.